src/swarm/neo/client/mixins/SuspendableRequestCore.d

```diff
@@ -0,0 +1,1165 @@
+    scope class ExampleRequestImpl
+    {
+        import swarm.neo.connection.RequestOnConnBase;
+
+        private RequestOnConn.EventDispatcherAllNodes conn;
+        private ExampleRequest.Context* context;
+
+        public this ( RequestOnConn.EventDispatcherAllNodes conn,
+            ExampleRequest.Context* context )
+        {
+            this.conn = conn;
+            this.context = context;
+        }
+
+        // Instantiates a SuspendableRequestInitialiser with dummy delegates
+        // passed as the various policies.
+        public void run ( )
+        {
+            auto initialiser = createSuspendableRequestInitialiser!(ExampleRequest)(
+                this.conn, this.context, &this.fillPayload,
+                &this.handleStatusCode);
+
+            // Pass initialiser to an AllNodesRequest...
+        }
+
+        // Dummy policies...
+        private void fillPayload (
+            RequestOnConnBase.EventDispatcher.Payload payload ) { }
+
+        private bool handleStatusCode ( ubyte status )
+        {
+            return true;
+        }
+    }
+}
+
+/*******************************************************************************
+
+    Helper function to instantiate AllNodesRequest with a
+    SuspendableRequestInitialiser. Using this function allows the compiler to
+    infer the template arguments of AllNodesRequest and
+    SuspendableRequestInitialiser from the types of the function arguments.
+
+    Params:
+        Request = type of the request being handled
+        Connector = type of policy instance to be called to perform connection
+            establishment (see AllNodesRequest)
+        Disconnected = type of policy instance to be called when an I/O error
+            causes the connection to the node to break (see AllNodesRequest)
+        FillPayload = type of policy instance to be called to add any required
+            fields to the initial message payload sent to the node
+        HandleStatusCode = type of policy instance to be called to validate the
+            status code received from the node
+        Handler = type of policy instance to be called to handle the main logic
+            of the request (see AllNodesRequest)
+        conn = request-on-conn event dispatcher to use for handling the request
+        context = request context
+        connector = instance of Connector policy
+        disconnected = instance of Disconnected policy
+        fill_payload = instance of FillPayload
+        handle_status_code = instance of HandleStatusCode
+        handler = instance of Handler policy
+
+    Returns:
+        instance of AllNodesRequest constructed with the provided arguments
+
+*******************************************************************************/
+
+public AllNodesRequest!(Request, Connector, Disconnected,
+    SuspendableRequestInitialiser!(Request, FillPayload, HandleStatusCode),
+    Handler)
+    createSuspendableRequest ( Request, Connector, Disconnected, FillPayload,
```

Comment:
I have no better suggestion here, but it's very hard to spot the function's name with the current indentation (all these helper functions have the same problem, but this one is the worst).